include/swift/AST/DiagnosticsSema.def

```diff
@@ -127,35 +128,35 @@ ERROR(unexpected_arguments_in_contextual_member,none,
       "%0 %1 is not a function", (DescriptiveDeclKind, DeclName))
 
 ERROR(could_not_use_value_member,none,
-      "member %1 cannot be used on value of type %0", (Type, DeclName))
+      "member %1 cannot be used on value of type %0", (Type, DeclNameRef))
 ERROR(could_not_use_type_member,none,
-      "member %1 cannot be used on type %0", (Type, DeclName))
+      "member %1 cannot be used on type %0", (Type, DeclNameRef))
 
 ERROR(could_not_use_type_member_on_instance,none,
       "static member %1 cannot be used on instance of type %0",
-      (Type, DeclName))
+      (Type, DeclNameRef))
 ERROR(could_not_use_enum_element_on_instance,none,
       "enum case %0 cannot be used as an instance member",
-      (DeclName))
+      (DeclNameRef))
 ERROR(could_not_use_type_member_on_protocol_metatype,none,
       "static member %1 cannot be used on protocol metatype %0",
-      (Type, DeclName))
+      (Type, DeclNameRef))
 ERROR(could_not_use_instance_member_on_type,none,
       "instance member %1"
       "%select{| of type %2}3 cannot be used on"
       "%select{| instance of nested}3 type %0",
-      (Type, DeclName, Type, bool))
+      (Type, DeclNameRef, Type, bool))
 ERROR(could_not_use_member_on_existential,none,
       "member %1 cannot be used on value of protocol type %0; use a generic"
       " constraint instead",
-      (Type, DeclName))
+      (Type, DeclNameRef))
 FIXIT(replace_with_type,"%0",(Type))
 FIXIT(insert_type_qualification,"%0.",(Type))
 
 ERROR(candidate_inaccessible,none,
       "%0 is inaccessible due to "
       "'%select{private|fileprivate|internal|%error|%error}1' protection level",
-      (DeclName, AccessLevel))
+      (DeclBaseName, AccessLevel))
```

Comment:
These are all cases where we actually found a declaration. Should we use the declaration’s actual name instead of the name it was looked up with?